### ISSUE TITLE
Remove guards for non-existent speech view

### ISSUE THREAD

**Issue by morganwillcock** [adventuregamestudio/ags#1828]
Running a game made in AGS 2.52 causes the engine to exit when a Character attempts to speak but has no speech view. I've checked in the version 2.52 of the Editor, the Normal view cannot be unset but the Talking view is not mandatory for speech to work.

Character setup:
![Character](https://user-images.githubusercontent.com/14958747/201476190-28f6b4a5-5e0b-4011-9c05-35b89559a9a3.png)

Room script:
```ags
function room_a() {
  // script for room: Player enters screen (after fadein)
  DisplaySpeech(EGO, "Testing speech");
}
```

Result in 2.52 engine:
![Game](https://user-images.githubusercontent.com/14958747/201476232-1722a98b-9555-45cc-9d05-d650f1c0643d.png)

This is not a fallback to View 1. If the script is modified so that the Character is not using View 1 at the time it speaks:
```ags
function room_a() {
  // script for room: Player enters screen (after fadein)
  SetCharacterView(EGO, 2);
  DisplaySpeech(EGO, "Testing speech");
}
```
...the active View remains and is not reverted to View 1.

**Comment by ivan-mogilko** [comment]
I tested this with the recent 3.6.0, and do not experience any problems, the engine seem to have same behavior as you mentioned above: it keeps the normal view if there's no speech view.

Please clarify, which engine version (or commit) have you used, and maybe attach a game (or post a link to game) that causes issues?

**Comment by morganwillcock** [comment]
It is this game: http://www.screen7.co.uk/games/download/poom_full.zip

Data version is actually 2.56; I mistakenly used a slightly older Editor when testing what the older Editor did.

I've re-tested it on the latest commit (a4a98adfcb039e0f8f213d60e8b0b639b0217f9b) and see the same problem.

Here are steps to reproduce it with [this saved game](https://github.com/adventuregamestudio/ags/files/10050675/agssave.002.zip):
1. Move cursor to top of screen
2. Left-click on inventory icon (the paper bag)
3. Left-click on the second item in the inventory (the "icebreaker" cartridge)
4. Left-click OK button
5. Right click on "Controls" in the room

```
Quitting the game...
An error has occurred. Please contact the game author for support, as this is likely to be an error in game logic or script and not a bug in AGS engine.
(ACI version 3.6.0.37)


Error: Character cLloyd speech view 13 does not have necessary loop 3 or it has no frames; and there's no valid loop to fall back.
```
If you look at the code it is a little confusing because it is re-using an existing character as a new character which is invisible (the voice is meant to be coming from the computer console which is already visible).

For testing purposes, I avoided the crash by doing this:
```diff
diff --git a/Engine/ac/character.cpp b/Engine/ac/character.cpp
index bba2efddc..73331722f 100644
--- a/Engine/ac/character.cpp
+++ b/Engine/ac/character.cpp
@@ -2540,7 +2540,7 @@ void _displayspeech(const char*texx, int aschar, int xx, int yy, int widd, int i
                 speakingChar->scrname, speakingChar->view + 1, speakingChar->loop);
             // is there even a fallback loop?
             if (views[speakingChar->view].numLoops == 0 || views[speakingChar->view].loops[0].numFrames == 0)
-                quitprintf("!%s; and there's no valid loop to fall back.", err.GetCStr());
+                debug_script_warn("!%s; and there's no valid loop to fall back.", err.GetCStr());
             debug_script_warn("WARNING: %s; switching to loop 0.", err.GetCStr());
             speakingChar->loop = 0;
         }
@@ -2800,7 +2800,7 @@ void _displayspeech(const char*texx, int aschar, int xx, int yy, int widd, int i
                     speakingChar->scrname, speakingChar->view + 1, speakingChar->loop);
                 // is there even a fallback loop?
                 if (views[speakingChar->view].numLoops == 0 || views[speakingChar->view].loops[0].numFrames == 0)
-                    quitprintf("!%s; and there's no valid loop to fall back.", err.GetCStr());
+                    debug_script_warn("!%s; and there's no valid loop to fall back.", err.GetCStr());
                 debug_script_warn("WARNING: %s; switching to loop 0.", err.GetCStr());
                 speakingChar->loop = 0;
             }
```
...although I imagine this probably isn't safe to do

**Comment by ivan-mogilko** [comment]
I observe the error in this game, but I'm also puzzled now, because when checking under debugger this character has a speech view assigned. How is this related to the problem mentioned in this ticket?

**Comment by ivan-mogilko** [comment]
I tested the old editor a bit more, and, apparently, when the speech loop is empty, the old engine was simply using the dummy frame from the same loop (AGS creates a dummy frame for each empty loop to prevent the crashes) without falling back to any other loop or view.

The curious thing also is that there's a bug in the old 2.5* editors, where if you create a loop and then delete all frames one by one, the dummy frame would have the last remembered sprite from that loop. Probably the last frame's data is simply left in memory when you delete frames in the editor.

**Comment by ivan-mogilko** [comment]
In regards to the problem mentioned in the first post, I found out that the error occurs when the speech view is assigned, but has no frames in loops whatsoever. The original engine kept working, displaying dummy frame instead, but 3.6.0 quits with error.

**Comment by ivan-mogilko** [comment]
Pushed a fix: 95c99fc325f8550f220e7a0f67d6555bb23c8c5a

**Comment by morganwillcock** [comment]
Thanks, I'll try it out. Sorry for muddling up the issues in the ticket.
